src/ToolInfo.php

```diff
@@ -26,67 +26,92 @@
     const COMPOSER_INSTALLED_FILE = '/../../composer/installed.json';
     const COMPOSER_PACKAGE_NAME = 'friendsofphp/php-cs-fixer';
 
-    public static function getComposerVersion()
+    /**
+     * @var string
+     */
+    private $script;
+
+    /**
+     * @var null|string
+     */
+    private $composerVersion;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledAsPhar;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledByComposer;
+
+    /**
+     * @var null|string
+     */
+    private $scriptDir;
+
+    /**
+     * @param ?string $script
+     */
+    public function __construct($script = null)
     {
```

Comment:
I prefer non-optional parameter too, but only in a well-organized dependency injection context.
As of yet, ToolInfo needs to be instantiade where it is used.